store/core/src/main/java/org/apache/carbondata/store/master/Master.java

```diff
@@ -0,0 +1,530 @@
+    String tablePath = getTableFolder(database, tableName);
+    CarbonTable carbonTable;
+    SoftReference<CarbonTable> reference = cacheTables.get(tablePath);
+    if (reference != null) {
+      carbonTable = reference.get();
+      if (carbonTable != null) {
+        return carbonTable;
+      }
+    }
+
+    try {
+      org.apache.carbondata.format.TableInfo tableInfo =
+          CarbonUtil.readSchemaFile(CarbonTablePath.getSchemaFilePath(tablePath));
+      SchemaConverter schemaConverter = new ThriftWrapperSchemaConverterImpl();
+      TableInfo tableInfo1 = schemaConverter.fromExternalToWrapperTableInfo(tableInfo, "", "", "");
+      tableInfo1.setTablePath(tablePath);
+      carbonTable = CarbonTable.buildFromTableInfo(tableInfo1);
+      cacheTables.put(tablePath, new SoftReference<CarbonTable>(carbonTable));
+      return carbonTable;
+    } catch (IOException e) {
+      String message = "Failed to get table from " + tablePath;
+      LOGGER.error(e, message);
+      throw new StoreException(message);
+    }
+  }
+
+  public boolean createTable(TableInfo tableInfo, boolean ifNotExists) throws IOException {
+    AbsoluteTableIdentifier identifier = tableInfo.getOrCreateAbsoluteTableIdentifier();
+    boolean tableExists = FileFactory.isFileExist(identifier.getTablePath());
+    if (tableExists) {
+      if (ifNotExists) {
+        return true;
+      } else {
+        throw new IOException(
+            "car't create table " + tableInfo.getDatabaseName() + "." + tableInfo.getFactTable()
+                .getTableName() + ", because it already exists");
+      }
+    }
+
+    SchemaConverter schemaConverter = new ThriftWrapperSchemaConverterImpl();
+    String databaseName = tableInfo.getDatabaseName();
+    String tableName = tableInfo.getFactTable().getTableName();
+    org.apache.carbondata.format.TableInfo thriftTableInfo =
+        schemaConverter.fromWrapperToExternalTableInfo(tableInfo, databaseName, tableName);
+
+    String schemaFilePath = CarbonTablePath.getSchemaFilePath(identifier.getTablePath());
+    String schemaMetadataPath = CarbonTablePath.getFolderContainingFile(schemaFilePath);
+    FileFactory.FileType fileType = FileFactory.getFileType(schemaMetadataPath);
+    try {
+      if (!FileFactory.isFileExist(schemaMetadataPath, fileType)) {
+        boolean isDirCreated = FileFactory.mkdirs(schemaMetadataPath, fileType);
+        if (!isDirCreated) {
+          throw new IOException("Failed to create the metadata directory " + schemaMetadataPath);
+        }
+      }
+      ThriftWriter thriftWriter = new ThriftWriter(schemaFilePath, false);
+      thriftWriter.open(FileWriteOperation.OVERWRITE);
+      thriftWriter.write(thriftTableInfo);
+      thriftWriter.close();
+      return true;
+    } catch (IOException e) {
+      LOGGER.error(e, "Failed to handle create table");
+      throw e;
+    }
+  }
+
+  public void openSegment(CarbonLoadModel loadModel, boolean isOverwriteTable) throws IOException {
```

Comment:
Please make the visibility minimum, as of now, I think it is better to provide createTable, loadData, etc only in Master